src/realm/group_writer.cpp

```diff
@@ -687,8 +687,12 @@ std::pair<size_t, size_t> GroupWriter::extend_free_space(size_t requested_size)
     // extended the file size. It can also happen as part of initial file expansion
     // during attach_file().
     size_t logical_file_size = to_size_t(m_group.m_top.get(2) / 2);
-    size_t extend_size = requested_size;
-    size_t new_file_size = logical_file_size + extend_size;
+    size_t new_file_size = logical_file_size;
+    if (REALM_UNLIKELY(int_add_with_overflow_detect(new_file_size, requested_size))) {
```

Comment:
@finnschiermer though it was removed in 82f7c7ebc13493da1a352205e1dac3c4933c86df I think this check is necessary here so I added it back. If I am mistaken can you please explain?